Sources/_Runtime/Metadata/ClassMetadata.swift

```diff
@@ -0,0 +1,52 @@
+//===----------------------------------------------------------*- swift -*-===//
+//
+// This source file is part of the Swift Argument Parser open source project
+//
+// Copyright (c) 2021 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+//
+//===----------------------------------------------------------------------===//
+
+#if _ptrauth(_arm64e)
+import _CRuntime
+#endif
+
+public struct ClassMetadata: Metadata, PointerView {
+  typealias View = _ClassMetadata
+
+  public let pointer: UnsafeRawPointer
+
+  var descriptor: ContextDescriptor {
+    // Type descriptors are signed on arm64e using pointer authentication.
+    #if _ptrauth(_arm64e)
+    let signed = __ptrauth_strip_asda(view._descriptor.pointer)
+    return ContextDescriptor(pointer: signed)
+    #else
+    return view._descriptor
+    #endif
+  }
+}
+
+struct _ClassMetadata {
```

Comment:
These `View` declaration need to move into `_CRuntime` — Swift doesn’t guarantee type layout, so there’s no way to know / ensure that these will match the runtime’s metadata layout.

I think computed properties like `ClassMetadata.descriptor`, above, are the right way to go for anything where you need an actual type instead of an int or a pointer.

Reply:
Argh, you're so right. Really wishing we had a `@repr(c)` attribute or similar in Swift!